lib/node_modules/@stdlib/string/len/lib/next-break.js

```diff
@@ -0,0 +1,98 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2020 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var codePointAt = require( './code-point-at' );
+var isSurrogate = require( './is-surrogate' );
+var shouldBreak = require( './should-break' );
+var getGraphemeBreakProperty = require( './get-grapheme-break-property.js' );
+var getEmojiProperty = require( './get-emoji-property.js' );
+
+
+// MAIN //
+
+/**
+* Returns the next grapheme break in the string after the given index.
+*
+* @param {string} str - input string
+* @param {number} idx - position in string for starting search
+* @returns {number} next grapheme break position
+*
+* @example
+* var out = nextBreak( 'last man standing', 4 );
+* // returns 5.0
+*
+* @example
+* var out = nextBreak( 'presidential election', 8 );
+* // returns 9.0
+*
+* @example
+* var out = nextBreak( 'अनुच्छेद', 1 );
+* // returns 3.0
+*
+* @example
+* var out = nextBreak( '🌷', 0 );
+* // returns 2.0
+*/
+function nextBreak( str, idx ) {
+	var prevEmoji;
+	var nextEmoji;
+	var midEmoji;
+	var prevCP;
+	var nextCP;
+	var prev;
+	var next;
+	var mid;
+	var i;
+	if ( !idx ) {
+		idx = 0;
+	}
+	if ( idx < 0 ) {
+		return 0;
+	}
+	if ( idx >= str.length - 1 ) {
```

Comment:
Given that `str.length` is used repeatedly in this function, you can probably go ahead and cache in another variable.